### PR TITLE
Bug fix to Stack repr which also broke resp repr

### PR DESCRIPTION
**Pull request by ehsteve**

### CODE REVIEW

File: roentgen/absorption/material.py [hunk 1]
```diff
@@ -291,17 +291,23 @@ def __init__(self, optical_path, detector):
...
         txt = "Response(path="
-        for material in self.optical_path:
-            txt += str(material)
-        txt += " detector=" + str(self.detector)
+        if isinstance(self.optical_path, Material):
+            txt += f"{self.optical_path} "
+        else:  # must be a Stack
+            for material in self.optical_path.materials:
+                txt += f"{material} "
+        txt += f", detector={self.detector})"
```
Comment (samaloney) [inline roentgen/absorption/material.py:299]: If optical path has to be either a Material or Stack but if they both have reprs could this not be:
```
txt = f"Response(path={self.optical_path},  detector={self.detector})"
```

File: roentgen/absorption/material.py [hunk 2]
```diff
@@ -291,17 +291,23 @@ def __init__(self, optical_path, detector):
...
         return txt + ")"
 
     def __str__(self):
         """Returns a human-readable representation."""
-        txt = "path="
-        for material in self.optical_path:
-            txt += str(material) + " "
-        txt += " detector=" + str(self.detector)
+        txt = "Response(path="
```
Comment (samaloney) [inline roentgen/absorption/material.py:304]: Same comment as above?

**Review by samaloney — COMMENTED** [review]
Looks good prob more of `__str___` as you can't directly use the output from the repr in back in python which would be very nice.

Current
```
mat = Material('air', 1 * u.m)

mat
Material(Air (dry)) 1.0 m 1.2 kg / m3)
```

Nice to have something like this
```
mat = Material('air', 1 * u.m)

mat
Material('air', thickness = 1 * u.m, density=1.2*Unit("g / cm3"))
```

**Comment by ehsteve** [comment]
Thanks. I'll fix. I never understand the difference between repr and str so if you have any suggestions let me know.

File: roentgen/absorption/material.py (hunk 1 above)
Comment (ehsteve) [reply roentgen/absorption/material.py:299]: Good point!

**Review by samaloney — APPROVED** [review]

**Pull request merged**